### PR TITLE
[NFC] Hide SourceFile::Decls

### PR DESCRIPTION
**Pull request by CodaFi**
In preparation for installing some stable paths infrastructure here,
hide access to the array of top-level decls.

### CODE REVIEW

**Comment by CodaFi** [comment]
@swift-ci please smoke test and merge

**Comment by davidungar** [comment]
Nice!

**Review by davidungar — APPROVED** [review]
LGTM!

File: include/swift/AST/SourceFile.h [hunk 1]
```diff
@@ -125,11 +125,28 @@ class SourceFile final : public FileUnit {
...
+  std::vector<Decl *> Decls;
+
   friend ASTContext;
   friend Impl;
+
 public:
-  /// The list of top-level declarations in the source file.
-  std::vector<Decl*> Decls;
+  /// Appends the given declaration to the end of the top-level decls list.
+  void addTopLevelDecl(Decl *d) {
```
Comment (slavapestov) [inline include/swift/AST/SourceFile.h]: Eventually this should set them all in one shot so that anything depending on getTopLevelDecls() is immutable.

**Review by slavapestov — APPROVED** [review]
Nice cleanup!

**Comment by CodaFi** [comment]
Urf, need LLDB patches.

**Comment by CodaFi** [comment]
apple/llvm-project#525

@swift-ci smoke test

File: include/swift/AST/SourceFile.h (hunk 1 above)
Comment (CodaFi) [reply include/swift/AST/SourceFile.h]: LLDB is going to be the last bastion of mutations here...

**Comment by CodaFi** [comment]
apple/llvm-project#525

@swift-ci smoke test

**Comment by CodaFi** [comment]
apple/llvm-project#525

@swift-ci smoke test

**Comment by CodaFi** [comment]
apple/llvm-project#525

@swift-ci smoke test

**Comment by CodaFi** [comment]
Installed a workaround for LLDB and filed rdar://58355191 about removing it on their end.

**Comment by CodaFi** [comment]
apple/llvm-project#525

@swift-ci smoke test

**Comment by CodaFi** [comment]
apple/llvm-project#525

@swift-ci smoke test

**Comment by CodaFi** [comment]
⛵️

**Pull request merged**